src/main/webapp/courseLoader.js

```diff
@@ -1,164 +1,205 @@
+    fetch(
+      `/term-live?school-name=${schoolName}&course-name=${courseName}&term=${term}&prof-name=${profName}&num-units=${units}`
+    )
+      .then((response) => response.json())
+      .then((data) => {
+        termData = data;
+        makeGraphs(termData);
+      });
+  }
+
+  document.addEventListener("DOMContentLoaded", function () {
+    fillTitles();
+    populateData();
+  });
+
+  function makeGraphs(termDataObject) {
+    const termCommentsList = termDataObject.termCommentsList;
+    const dummyComments = ["dummy comment 1", "dummy comment 2"].concat(
+      termCommentsList
+    );
+    loadComments(dummyComments, /* true means course comments */ true);
+    loadComments(dummyComments, /* false means professor comments */ false);
+
+    const tempHoursList = termDataObject.hoursList;
+    const hoursList = [
+      ["hours"],
+      /* dummyHourRating */ [3],
+      /* dummyHourRating */ [8],
+    ].concat(tempHoursList);
+
+    const hourData = new google.visualization.arrayToDataTable(hoursList);
+    const hourOptions = {
+      title: "Hours Spent per Week",
+      legend: { position: "none" },
+      vAxis: { title: "# Students" },
+      hAxis: { title: "Hours" },
+      histogram: {
+        hideBucketItems: true,
+      },
+    };
+    const hourChart = new google.visualization.Histogram(
+      document.getElementById("hours-chart")
+    );
+    hourChart.draw(hourData, hourOptions);
+
+    const tempDiffList = termDataObject.difficultyList;
+    const diffList = [
+      ["difficulty"],
+      /* dummyDifficultyRating */ [1],
+      /* dummyDifficultyRating */ [4],
+    ].concat(tempDiffList);
+    const diffData = new google.visualization.arrayToDataTable(diffList);
+    const diffOptions = {
+      title: "Difficulty of Class",
+      legend: { position: "none" },
+      vAxis: { title: "# Students" },
+      hAxis: { title: "Difficulty" },
+      histogram: {
+        hideBucketItems: true,
+      },
+    };
+    const diffChart = new google.visualization.Histogram(
+      document.getElementById("diff-chart")
+    );
+    diffChart.draw(diffData, diffOptions);
+
+    const tempTermPerceptionList = termDataObject.termPerceptionList;
+    const termPerceptionList = [
+      ["Term Perception"],
+      /* dummyPerceptionRating */ [11],
+      /* dummyPerceptionRating */ [5],
+    ].concat(tempTermPerceptionList);
+    const termPerceptionData = new google.visualization.arrayToDataTable(
+      termPerceptionList
+    );
+    const termPerceptionOptions = {
+      title: "Perception of Term Reviews",
+      legend: { position: "none" },
+      vAxis: { title: "Perception" },
+      hAxis: { title: "Comment Quantity" },
+      histogram: {
+        hideBucketItems: true,
+      },
+    };
+    const termPerceptionChart = new google.visualization.Histogram(
+      document.getElementById("term-chart")
+    );
+    termPerceptionChart.draw(termPerceptionData, termPerceptionOptions);
+
+    makeTermRatingChart();
```

Comment:
Nit: Do you have to await this? I honestly have no idea so if it works right now then it's probably fine, but figured I'd point out

Reply:
I probably should, but it is working right now

Reply:
It works now because you don't need to have anything happen after the async function returns/etc.

However it's good practice to always return the async promise if you're doing something async, so I'd return this call instead of just calling it.

That way if whatever calls makeGraphs() in the future needs to do something afterwards it can just await or .then() it.

This also makes testing easier if you're testing the public API but the private function happens to be async.